src/test/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4KitTest.kt

```diff
@@ -123,6 +140,502 @@ class GoogleAnalyticsFirebaseGA4KitTest {
+            .addGDPRConsentState("Marketing", marketingConsent)
+            .addGDPRConsentState("Performance", performanceConsent)
+            .build()
+        filteredMParticleUser = FilteredMParticleUser.getInstance(user, kitInstance)
+
+        kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)
+
+
+        TestCase.assertEquals(0, firebaseSdk.getConsentState().size)
+    }
+    @Test
+    fun onConsentStateUpdatedTest_When_default_is_Unspecified() {
+        val map = HashMap<String, String>()
+        map["defaultAdStorageConsentSDK"] = "Unspecified"
+        map["defaultAnalyticsStorageConsentSDK"] = "Unspecified"
+        map["consentMappingSDK"] =
+            "[{\\\"jsmap\\\":null,\\\"map\\\":\\\"Performance\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_user_data\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"Marketing\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_personalization\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"testconsent\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_storage\\\"}]"
+        map["defaultAdUserDataConsentSDK"] = "Unspecified"
+        map["defaultAdPersonalizationConsentSDK"] = "Unspecified"
+
+        kitInstance.configuration =
+            KitConfiguration.createKitConfiguration(JSONObject().put("as", map.toMutableMap()))
+
+        val marketingConsent = GDPRConsent.builder(true)
+            .document("Test consent")
+            .location("17 Cherry Tree Lane")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+
+        val performanceConsent = GDPRConsent.builder(true)
+            .document("parental_consent_agreement_v2")
+            .location("17 Cherry Tree Lan 3")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+        val state = ConsentState.builder()
+            .addGDPRConsentState("Marketing", marketingConsent)
+            .addGDPRConsentState("Performance", performanceConsent)
+            .build()
+        filteredMParticleUser = FilteredMParticleUser.getInstance(user, kitInstance)
+
+        kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)
+
+        val expectedConsentValue =
+            firebaseSdk.getConsentState().getKeyByValue("AD_USER_DATA").toString()
+        TestCase.assertEquals("GRANTED", expectedConsentValue)
+        val expectedConsentValue2 =
+            firebaseSdk.getConsentState().getKeyByValue("AD_PERSONALIZATION").toString()
+        TestCase.assertEquals("GRANTED", expectedConsentValue2)
+        TestCase.assertEquals(2, firebaseSdk.getConsentState().size)
+    }
+
+    fun  MutableMap<Any, Any>.getKeyByValue(inputKey: String): Any? {
+        for ((key, mapValue) in entries) {
+            if (key.toString() == inputKey) {
+                return mapValue
+            }
+        }
+        return null
+    }
+
+    @Test
+    fun testParseToNestedMap_When_JSON_Is_INVALID(){
```

Comment:
I noticed that you have tests here for cases where the JSON is invalid or empty, but I don't see a test for the "happy path" where it correctly parses JSON. Is that not necessary?